### PR TITLE
fix: add in filter and use action param for countAggregate

### PR DESCRIPTION
**Pull request by kaliang1**
This PR fixes the following issues with countAggregate: 
- In the SQL query, the array for the IN condition was not being set properly
- The action was using `QueryParam` instead of `ActionParam`

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/EbeanLocalDAO.java [hunk 1]
```diff
@@ -1214,6 +1214,7 @@ private static String constructCountAggregateSQLQuery(@Nonnull IndexCriterionArr
       }
       whereClause.append(" t").append(i).append(".aspect = ?");
       if (criterion.getPathParams() != null) {
+        validateConditionAndValue(criterion);
```
Comment (jywadhwani) [inline dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/EbeanLocalDAO.java:1217]: Probably missed it earlier, let's move this outside, we can check for `getParams` to be non-null in `validateConditionAndValue` method itself? This way you will fail early.

File: restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java [hunk 2]
```diff
@@ -560,8 +560,8 @@ private IndexFilter getDefaultIndexFilter() {
   @Action(name = ACTION_COUNT_AGGREGATE)
```
Comment (jywadhwani) [inline restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java:560]: I wonder why we didn't use a `Finder` method and return `CollectionResult<V, MetaData>`? I understand we are returning the count along with the field, but can we capture this map in the corresponding metadata?
Would love to hear your thoughts.

File: dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/EbeanLocalDAO.java (hunk 1 above)
Comment (kaliang1) [reply dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/EbeanLocalDAO.java:1217]: I think the `pathParams` can be null in the index criterion: https://github.com/linkedin/datahub-gma/blob/master/dao-api/src/main/pegasus/com/linkedin/metadata/query/IndexCriterion.pdl#L16

File: restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java (hunk 2 above)
Comment (kaliang1) [reply restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java:589]: That makes a lot of sense! I deprecated the old action and created a new finder.

File: dao-api/src/main/pegasus/com/linkedin/metadata/query/CountAggregateMetadata.pdl [hunk 3]
```diff
@@ -0,0 +1,11 @@
+namespace com.linkedin.metadata.query
+
+/**
+ * The model for count aggregate metadata
+ */
+record CountAggregateMetadata {
```
Comment (jywadhwani) [inline dao-api/src/main/pegasus/com/linkedin/metadata/query/CountAggregateMetadata.pdl:6]: how about we just call this `MapMetadata` - then we can reuse this for other purposes and not just aggregated results?

File: dao-api/src/main/pegasus/com/linkedin/metadata/query/CountAggregateMetadata.pdl [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  /**
+   * Map of the field values to the count
+   */
+  countAggregateMap: map[string, long]
```
Comment (jywadhwani) [inline dao-api/src/main/pegasus/com/linkedin/metadata/query/CountAggregateMetadata.pdl:10]: rename this field to something more generic, like `longMap` or something?

File: dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/EbeanLocalDAO.java (hunk 1 above)
Comment (jywadhwani) [reply dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/EbeanLocalDAO.java:1217]: Yes but we can check for non-null value inside `validateConditionAndValue` method itself?

Comment (jywadhwani) [reply dao-impl/ebean-dao/src/main/java/com/linkedin/metadata/dao/EbeanLocalDAO.java:1217]: We can address this separately.

**Review by jywadhwani — APPROVED** [review]

**Pull request merged**